File: Data Structures & Algorithms/longest-palindromic-substring/submission-1.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        ans = ""
        n = len(s)
        for i in range(0, n):
            left = i
            right = i + 1
            while left >= 0 and right < n and s[left] == s[right]:
                left -= 1
                right += 1
            if right - left - 1 > len(ans):
                ans = s[left+1: right]
            left = i - 1
            right = i + 1
            while left >= 0 and right < n and s[left] == s[right]:
                left -= 1
                right += 1
            if right - left - 1 > len(ans):
                ans = s[left+1: right]
        return ans

File: Data Structures & Algorithms/queue/submission-1.kt
class Deque {

    private var left: DoubleNode? = null
    private var right: DoubleNode? = null

    fun isEmpty(): Boolean {
        return left == null
    }

    fun append(value: Int) {
        val newNode = DoubleNode(value)
        if (isEmpty()) {
            left = newNode
            right = newNode
        } else {
            right!!.right = newNode
            newNode.left = right
            right = newNode
        }
    }

    fun appendleft(value: Int) {
        val newNode = DoubleNode(value)
        if (isEmpty()) {
            left = newNode
            right = newNode
        } else {
            left!!.left = newNode
            newNode.right = left
            left = newNode
        }
    }

    fun pop(): Int {
        val ans = right?.value ?: -1
        if (left == right) {
            left = null
            right = null
        } else {
            right = right?.left
            right!!.right = null
        }
        return ans
    }

    fun popleft(): Int {
        val ans = left?.value ?: -1
        if (left == right) {
            left = null
            right = null
        } else {
            left = left?.right
            left!!.left = null
        }
        return ans
    }
}


private class DoubleNode(val value: Int) {
    var left: DoubleNode? = null
    var right: DoubleNode? = null
}